include/net/net_offload.h

```diff
@@ -29,6 +29,25 @@ extern "C" {
 
 #if defined(CONFIG_NET_OFFLOAD)
 
+#ifndef CONFIG_LEGACY_TIMEOUT_API
+
+/** @cond INTERNAL_HIDDEN */
+
+static inline int32_t k_timeout_to_int32(k_timeout_t timeout)
```

Comment:
Is it ok to have a `k_` prefixed function in here? Isn't it reserved for public kernel APIs? @jukkar

Reply:
I would say this is ok in this case, after all we are converting from `k_timeout` to `int32` and the function name tells that.